### ISSUE TITLE
Fix Error When Compressing Static Files Through Gunicorn

### ISSUE THREAD

**Issue by nickdibari** [Moody-Tunes/moodytunes#254]
Once a month, we seem to get an error from the Django application. The error reads
 ```
PermissionError at /accounts/login/ [Errno 13] Permission denied: ‘/home/ansible/moodytunes/_static/CACHE/css/c5818cc19d4e.css’
```

This happens because the django-compressor plugin needs to rewrite the static files to the STATIC_DIR directory, and the user running the gunicorn process does not have permission to write to that directory. This is fixed by re provisioning the host, thereby updating the cache with the most recent compressed version of the file. However, since nothing about the file has changed we'd rather not need to maintain the box by provisioning the host once a month. We should find a way to resolve these errors so that the error no longer appears.

Two potential ways to solve this problem:

1. Make the `_static` directory writable by the `system` group. This will allow the gunicorn user to write to the directory when the static files need updating. This is not the best choice because it could allow the ability for the user to add files to the directory, but it is the easiest one.

2. Make the compressor plugin not update files on a timeout. This would stop the plugin from trying to recompile the files every month and stop the application from trying to rewrite files in the first place. This is the better option because it removes the problem of needing to rewrite the files, but care should be taken that when static files change or new ones are added that the static files are recompiled

Deliverables
- Fix for the PermissionError generated when the gunicorn process tries to recompile static files

**Comment by nickdibari** [comment]
NOTE

Looks like the [COMPRESS_REBUILD_TIMEOUT](https://django-compressor.readthedocs.io/en/stable/settings/#django.conf.settings.COMPRESS_REBUILD_TIMEOUT) setting controls how long django-compressor should wait before rebuilding files. The default is set to 30 days, which matches up with the rate at which we've been seeing errors. Is there a way to set this so it will never rebuild?

**Comment by nickdibari** [comment]
SOLUTION

We should enable offline compression in production. Accord to [this guide](https://django-compressor.readthedocs.io/en/stable/behind-the-scenes/#offline-compression) when COMPRESS_OFFLINE is set to `True` no compressing will be done as part of the request/response loop. This setting assumes that the static files have been compressed ahead of time, and does not do any lookups or checking of the static files, it simply checks the manifest file outputted by the `compress` management command.

This setting should allow us to skip compressing files during the request and instead just use whatever values have already been set when the `compress` command has been run. Since this command is run as part of the production provisioning process we are guaranteed to have the latest compressed files generated in the compressor cache and set in the manifest. If we set the COMPRESS_OFFLINE value to `True` in production and leave it `False` in dev, this should (in theory) allow us to compress static files on the fly in development, deploy them to prod, and leave them alone. 

TL;DR: We're gonna use a setting that tells django-compressor to serve whatever files are currently in the cache and not worry about updating out-of-date files. We'll compress files as part of the deployment process to ensure that any static files that get changed are compressed and added to the manifest properly.
